modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -11,12 +12,78 @@ const WORLD_SIZE = 512;
 
 const defaultProps = {
   uniqueIdProperty: {type: 'string', value: ''},
-  highlightedFeatureId: null
+  highlightedFeatureId: null,
+  tileJSON: null,
+  onTileJSONLoad: {type: 'function', optional: true, value: null, compare: false}
 };
 
 export default class MVTLayer extends TileLayer {
+  initializeState() {
+    super.initializeState();
+    this.setState({
+      data: null,
+      tileJSON: null,
+      fetchingTileJSON: false
+    });
+  }
+
+  get isLoaded() {
+    const {fetchingTileJSON} = this.state;
+    return !fetchingTileJSON && super.isLoaded;
+  }
+
+  updateState({props, oldProps, context, changeFlags}) {
+    changeFlags.dataChanged = changeFlags.dataChanged || props.tileJSON !== oldProps.tileJSON;
+
+    if (changeFlags.dataChanged) {
+      // Save the fetchingTileJSON state - should not trigger a rerender
+      this.state.fetchingTileJSON = changeFlags.dataChanged;
+    }
+
+    super.updateState({props, oldProps, context, changeFlags});
+
+    if (changeFlags.dataChanged) {
+      this._updateTileData({props});
+    }
+  }
+
+  async _updateTileData({props}) {
+    const {onTileJSONLoad} = this.props;
+    const {tileset} = this.state;
+    let {data, tileJSON, minZoom, maxZoom} = props;
+
+    if (tileJSON) {
+      if (typeof tileJSON === 'string') {
```

Comment:
Can `tileJSON` be anything other than string? If not, declare it in defaultProps:

`tileJSON: {type: 'object', value: null, async: true}`

And skip the custom loading. You'll be able to check its status by

`this.internalState.isAsyncPropLoading('tileJSON')`